### PR TITLE
A couple of minor code cleanups

### PR DESCRIPTION
**Pull request by bangerth**
This updates recent commits #906. #907. #908.

### CODE REVIEW

File: include/deal.II/numerics/vector_tools.templates.h [hunk 1]
```diff
@@ -6724,11 +6724,8 @@ namespace VectorTools
                const InVector        &fe_function,
                const Point<spacedim>      &point)
   {
-#ifdef DEBUG
-    const FiniteElement<dim> &fe = dof.get_fe();
-    Assert(fe.n_components() == 1,
+    Assert(dof.get_fe().n_components() == 1,
            ExcMessage ("Finite element is not scalar as is necessary for this function"));
-#endif
```
Comment (tamiko) [inline include/deal.II/numerics/vector_tools.templates.h]: This is bikeshedding at its finest...

**Comment by tamiko** [comment]
But as you wish :-)

File: include/deal.II/numerics/vector_tools.templates.h (hunk 1 above)
Comment (tamiko) [inline include/deal.II/numerics/vector_tools.templates.h]: (Which by no means implies that you are not absolutely right with any of these changes.)

Comment (bangerth) [inline include/deal.II/numerics/vector_tools.templates.h]: I don't disagree with the bikeshedding comment. It's just that if you don't fix things that could be cleaned up right away, it never happens. It's not a comment on anyone else's changes. It's just regular maintenance.

Comment (bangerth) [inline include/deal.II/numerics/vector_tools.templates.h]: I also deleted 12 lines of code, and I think that's always the right direction ;-)

**Pull request merged**